src/main/java/io/nats/client/ConnectionImpl.java

```diff
@@ -1458,7 +1458,15 @@ void processMsg(byte[] data, int offset, int length) {
             // It's possible that we end up not using the message, but that's ok.
             Message msg = new Message(parser.ps.ma, sub, data, offset, length);
 
-            sub.lock();
+            MsgDeliveryWorker mdw = null;
+            if (sub instanceof AsyncSubscriptionImpl) {
+                mdw = ((AsyncSubscriptionImpl) sub).getDeliveryWorker();
+            }
+            if (mdw != null) {
```

Comment:
Is it possible to have a race on `mdw` and not be able to obtain the lock?

Reply:
No since the worker is assigned to the subscription when the sub is being created. From that point on it is immutable.